user-guide/Advanced_Modules/User Defined APIs/UserDefinableApiEndpoint.md

```diff
@@ -0,0 +1,175 @@
+        }
+      }
+    ],
+    "Enrich": [ "FromLogContext" ]
+  },
+
+  "UserDefinableAPIs": {
+    "NatsSubject": "Skyline.DataMiner.Protobuf.Apps.UserDefinableApis.Api.v1.UserDefinableApiTriggerRequest",
+    "SessionConfigPath": "C:\\Skyline DataMiner\\SLCloud.xml",
+    "CredentialsConfigPath": "C:\\Skyline DataMiner\\NATS\\nsc\\.nkeys\\creds\\DataMinerOperator\\DataMinerAccount\\DataMinerUser.creds",
+    "MessageBrokerTimeOutSeconds": 300
+  }
+}
+```
+
+### Changing a configuration
+
+If you want to change a configuration, create an appsettings.custom.json file, this will prevent your settings from being overwritten by an upgrade as explained earlier. You add the setting that you want to override, and change the value. In the example below, we are going to change the `MessageBrokertimeOutSeconds` to a higher value:
+
+`appsettings.custom.json`
+
+```json
+{
+  "UserDefinableAPIs": {
+    "MessageBrokerTimeOutSeconds": 500
+  }
+}
+```
+
+> [!NOTE]
+> For some of the configurations, a restart of the service is required when doing a change.
+
+### Kestrel
+
+Contains the endpoint on which UserDefinableApiEndpoint will be run, you can change the port here.
+
+IIS also has a rewrite rule (Reroute User Definable APIs) that forwards API requests to the port used by UserDefinableApiEndpoint (5002). When changing the port configured in appsettings(.custom).json, this rule also has to be updated.
+
+To update the rewrite rule, open `Internet Information Services (IIS) Manager` and follow the steps below:
+
+1. Open the 'Default Web Site' on the left;
+2. Click 'URL Rewrite';
+3. Click the 'Reroute User Definable APIs' rewrite rule;
+4. Click 'Edit' in the bar on the right;
+5. Change the default port (5002) in the 'Rewrite URL' under 'Action' to the port you want to use;
+6. Click 'Apply'
+
+![IIS Manager 1](~/user-guide/images/UDAPIS_IIS1.jpg)
+
+![IIS Manager 2](~/user-guide/images/UDAPIS_IIS2.jpg)
+
+![IIS Manager 3](~/user-guide/images/UDAPIS_IIS3.jpg)
+
+### Serilog
+
+Serilog is the logging service being used for UserDefinableApiEndpoint. Here you can change where the logfiles should be located, how big they can get and how many files should be kept. You can also change the default loglevels.
+
+> [!NOTE]
+> The default settings are to roll a file after it reaches a size of 5 MB. It will keep a maximum of 3 files. After reaching 3 files, the oldest file will be deleted to make place for the new file.
+
+### UserDefinableAPIs
+
+Specific options for User Definable APIs.
+
+#### NatsSubject
+
+The subject used for internal NATS messaging. This should not be changed.
+
+#### SessionConfigPath
+
+This is the path of the SLCLoud.xml file.
+
+#### CredentialsConfigPath
+
+This contains the path to the credentials used to connect to DataMiner over NATS.
+
+#### MessageBrokerTimeOutSeconds
+
+Default is set to 5 minutes (300 seconds). The MessageBroker sending the NATS trigger to SLNet will wait for this long on a response untill it will time out.
```

Comment:
```suggestion
Default is set to 5 minutes (300 seconds). The MessageBroker sending the NATS trigger to SLNet will wait for this long on a response until it will time out.
```